packages/cli/src/util/create-git-meta.ts

```diff
@@ -78,16 +78,20 @@ function getLastCommit(directory: string): Promise<git.Commit> {
 
 export function isDirty(directory: string, output: Output): Promise<boolean> {
   return new Promise(resolve => {
-    exec('git status -s', { cwd: directory }, function (err, stdout, stderr) {
-      let debugMessage = `Failed to determine if Git repo has been modified:`;
-      if (err || stderr) {
-        if (err) debugMessage += `\n${err}`;
-        if (stderr) debugMessage += `\n${stderr.trim()}`;
-        output.debug(debugMessage);
-        return resolve(false);
+    exec(
+      'git --no-optional-locks status -s',
```

Comment:
**suggestion (non-blocking):** add comment about why `--no-optional-locks` is necessary

Can we add a comment about this flag? The flag name and its [official description](https://git-scm.com/docs/git/2.22.1#Documentation/git.txt---no-optional-locks) don't do a great job explaining what the actual effects are.

Reply:
Good idea! Done!

Reply:
If we're worried about the order of the flags, we could use the `GIT_OPTIONAL_LOCKS=0` env var as an alternative